mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs

```diff
@@ -263,9 +255,18 @@ void ConstructCRef (StringBuilder sb)
 			if (DescKind == Kind.Type)
 				return;
 
+			if (ExplicitImplMember != null) {
+				sb.Append ('$');
```

Comment:
Uh...what?

The nice thing about using ECMA CREFs is that they already exist, there's already tooling for them (`csc /doc`, etc.), and thus there's a general "understanding" of what they represent.

The bad thing about ECMA CREFs is that they already exist, and thus "extending" them isn't necessarily a good idea, as it possibly compromises compatibility with existing toolchains or existing *readers*.

This is the latter: You're apparently using `$` to start the name of an explicit interface name, *but there's already a standard convention for explicit interface implementation*: `s/\//#/g`.

For example, in your build tree you should have `mono/mcs/tools/mdoc/Test/DocTest.xml`, which is the `mcs /doc` output when building DocTest.dll. DocTest.dll contains `MyList<A, B>`, which [explicitly implements `IEnumerable.GetEnumerator()`](https://github.com/mono/mono/blob/master/mcs/tools/mdoc/Test/DocTest-v1.cs#L677).

The resulting `csc /doc` XML?

        <member name="M:Mono.DocTest.Generic.MyList`2.System#Collections#IEnumerable#GetEnumerator">
            <remarks><c>M:Mono.DocTest.MyList`2.System#Collections#GetEnumerator</c>.</remarks></member>

The type, then, is:

    Mono.DocTest.Generic.MyList`2

while the member name is, weirdly-yet-understandably, `System#Collections#IEnumerable#GetEnumerator`.

This "falls out" of the ECMA CREF idea/requirement that for class members, everything from the last `.` to the end of the string is the member name, so a `string.Split()`-like implementation which only splits on `.` will get the correct member name. (This is *also* why constructors in `M` syntax use `#` instead of `.`, e.g. `M:Mono.DocTest.DocAttribute.#ctor(System.String)`.)

In short, using `$` is "weird" and wholly different from what an ECMA CREF should contain, resulting in strings such as `P:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item(System.Int32)` which should *really* be `P:System.Web.SessionState.HttpSessionStateContainer.System#Web#SessionState#IHttpSessionState#Item(System.Int32)`.